plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/editors/DateTimeInlineEditor.java

```diff
@@ -115,26 +113,30 @@ public void run() {
             if (parent.isDirty()) {
                 try {
                     Object value = parent.extractEditorValue();
-                    if (value instanceof Date || value == null) {
-                        editor.setValue((Date) value);
-                    }
+                    editor.setValue(value);
+                } catch (ClassCastException e) {
+                    DBWorkbench.getPlatformUI().showWarningMessageBox(ResultSetMessages.dialog_value_view_error_parsing_date_title,
```

Comment:
Never catch ClassCastException. Check type before cast.

Reply:
Replaced it with IllegalArgumentException